### ISSUE TITLE
Samsung Trend 2 green bitmap problem

### ISSUE THREAD

**Issue by mseroczynski** [commonsguy/cwac-camera#280]
Hey there,

I've got SM-G313HN here and it is currently the only device I've tested that has strange problem with taking photos. It looks like below:

**Comment by mseroczynski** [comment]
It also looks like this when i'm using demo app. 

If you guys have no idea what's going on please respond, I've got deadline on monday and even knowing that there's no clue may be very valuable information for me :)

**Comment by commonsguy** [comment]
Most likely, it is a preview size that the device advertises but does not actually support. However, there could be other things afoot. I do not have this device at present and so cannot provide you with a definitive solution.

**Comment by mseroczynski** [comment]
@commonsguy The thing is the preview is working correctly - it uses 480x800 which is correct devices resolution at full screen, this is not a preview screen, you knew it, right?

**Comment by commonsguy** [comment]
Sorry, it looked like a preview screen. Then it is some picture resolution that the device does not support. Various devices have that problem, which is why I have the whole `DeviceProfile` setup to be able to provide information about the maximum safe preview size. You are welcome to [create your own profile](https://github.com/commonsguy/cwac-camera#choosing-a-deviceprofile) for this device and try to come up with one that reliably gives you decent results.

**Comment by mseroczynski** [comment]
@commonsguy Thanks for your instant feedback :). Just tell me one thing - if i set in my Host custom picture size which is 100% correct and it's still not working - is there any possibility that creating DeviceProfile will fix my issue?

**Comment by commonsguy** [comment]
That is what I suggested in my last comment. What I do is buy the hardware (which you already have), then try all of the picture sizes and see which work. Usually, the problem is at one end of the range or the other, and of those, usually it is the higher-resolution pictures that exhibit problems. Then, create a `DeviceProfile` XML resource with the right name, and use `<maxPictureHeight>` to keep the `Camera.Size` values within a range that is known to be reliable.

**Comment by mseroczynski** [comment]
Ok, so if you're suggesting that this is a good way to i'm gonna check it out but please don't close this issue yet, i'll be back with report :)

**Comment by mseroczynski** [comment]
Well, providing *.xml file that is correctly recognized and loaded sets maxPictureHeight is not resolving my problem, is there any next step that you would suggest me :)?

**Comment by commonsguy** [comment]
No, sorry, I have no further suggestions, particularly in the timeframe that you are dealing with.

**Comment by mseroczynski** [comment]
Well, i guess the last solution without reimplementing whole module would be cutting one fourth of result bitmap on that particular device. I've checked othere DeviceProfile params like pictureDelay and so, none working obviously. Thank you for your time. 

Thread can be closed with "samsung-hit-the-fan" tag.

**Comment by commonsguy** [comment]
Are you saying that you tried every possible picture size, and they all exhibit this behavior?

I am going to leave the issue open, so I can eventually pick up a SM-G313HN and try to figure out what the problem is.

**Comment by mseroczynski** [comment]
I've tried most of them, but if all it does is making range smaller so in this CameraView.class line:

```
Size pictureSize = xact.host.getPictureSize(xact, pictureParams);
```

...code would assign max supported (in terms of profile) height, it makes no difference if i try "every single picture size" if its still there no matter if im defining:

```
	<maxPictureHeight>
1920
	</maxPictureHeight>
```

or

```
	<maxPictureHeight>
200
	</maxPictureHeight>
```

**Comment by commonsguy** [comment]
But your resulting picture is getting smaller? If the picture resolution is also unaffected, then the library is not picking up your profile. If the picture resolution is getting smaller *and* you have the same basic output as originally reported, then I don't know what to tell you. There's at least one Samsung model (one of the Ace line, IIRC) that has this sort of problem, and I have never found a solution for it.

**Comment by mseroczynski** [comment]
Result bitmap always has correct width and height (example maxheight = 1200 -> result is 1152 height), and always only 1/4th is filled up with pixels. Don't bother buying it, they'll only have one more reason to produce it :).

**Comment by mseroczynski** [comment]
Btw it seems like every other custom camera app is working correctly so it still may be worth looking into.

**Comment by droidekas** [comment]
Hi,I had a similar issue in Samsung baffinlitexx.Along with the issue that that the flash would not trigger.
The issue was solved by introducing a picture delay(~250) in the device profile.
Since my issue was only to get still pictures,the delay did not matter.
I presume @commonsguy  will add a suitable device profile after rigorous testing :).
I have also noticed that introducing a picture delay normally solves a lot of issues.
I agree this is not an exact solution but it could be an appropriate workaround instead of device profiles for the n number of devices available.

**Comment by commonsguy** [comment]
@droidekas 

> I presume @commonsguy will add a suitable device profile after rigorous testing :).

I assume that I am working on a replacement library and not doing anything of significance with this library.

> would you consider adding the custom profiles that users of this library end up making? That might ease up your work quite a bit.

I did in the past. As is noted on the project `README`, this library is slated to be replaced. I am working on the replacement now. My sincere hope is that device profiles will no longer be needed.
